pykoi/retrieval/vectordb/epsilla.py

```diff
@@ -0,0 +1,128 @@
+import os
+import numpy as np
+import types
+
+from typing import List, Union
+from langchain.embeddings.base import Embeddings
+from langchain.schema import BaseRetriever, Document
+from sklearn.decomposition import PCA
+from pyepsilla import vectordb
+
+from pykoi.retrieval.llm.constants import LlmName
+from pykoi.retrieval.vectordb.abs_vectordb import AbsVectorDb
+
+model_map = {
+    LlmName.OPENAI: 1536
```

Comment:
@goldmermaid in epsilla we define vector dimension on schema creation time, please let us know if this map makes sense, and does openai llm only provide 1536 dimension embedding model(s)

Reply:
Based on [doc](https://platform.openai.com/docs/guides/embeddings/what-are-embeddings), it looks like all openai v2 model has 1536 as embedding dimension while v1 model has different dimension and is not recommend for use.

qq: what will happen for epsilla if this defined vector dimension mismatch the embedding dimension?

Can we add a logic in the constructor to use corresponding `model` to map its dimension and through exception is model name is not a valid openai model?

Reply:
For the question "what will happen for epsilla if this defined vector dimension mismatch the embedding dimension?", the client will return error, we have the logic to throw the error. And for the `model` part, I think you guys have already handled the exception.